Error while checking build/O2/fullCI for 5f3709fa8fc0431180c66d76290deb9a60f1b21b at 2024-07-19 13:50:
```
## sw/BUILD/O2-latest/log
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
/sw/SOURCES/O2/13310-slc8_x86-64/0/DataFormats/common/include/CommonDataFormat/AbstractRef.h:34:14: error: expected '(' for function-style cast or type construction
/sw/SOURCES/O2/13310-slc8_x86-64/0/DataFormats/common/include/CommonDataFormat/AbstractRef.h:36:14: error: expected '(' for function-style cast or type construction
/sw/SOURCES/O2/13310-slc8_x86-64/0/DataFormats/common/include/CommonDataFormat/AbstractRef.h:38:14: error: expected '(' for function-style cast or type construction
/sw/SOURCES/O2/13310-slc8_x86-64/0/DataFormats/common/include/CommonDataFormat/AbstractRef.h:40:14: error: expected '(' for function-style cast or type construction
/sw/SOURCES/O2/13310-slc8_x86-64/0/DataFormats/common/include/CommonDataFormat/AbstractRef.h:50:27: error: program scope variable must reside in global or constant address space
/sw/SOURCES/O2/13310-slc8_x86-64/0/DataFormats/common/include/CommonDataFormat/AbstractRef.h:51:26: error: program scope variable must reside in global or constant address space
/sw/SOURCES/O2/13310-slc8_x86-64/0/DataFormats/common/include/CommonDataFormat/AbstractRef.h:52:26: error: program scope variable must reside in global or constant address space
/sw/SOURCES/O2/13310-slc8_x86-64/0/DataFormats/common/include/CommonDataFormat/AbstractRef.h:53:26: error: program scope variable must reside in global or constant address space
/sw/SOURCES/O2/13310-slc8_x86-64/0/DataFormats/common/include/CommonDataFormat/AbstractRef.h:60:29: error: argument may not have 'void' type
/sw/SOURCES/O2/13310-slc8_x86-64/0/DataFormats/common/include/CommonDataFormat/AbstractRef.h:61:30: error: argument may not have 'void' type
/sw/SOURCES/O2/13310-slc8_x86-64/0/DataFormats/common/include/CommonDataFormat/AbstractRef.h:65:31: error: argument may not have 'void' type
/sw/SOURCES/O2/13310-slc8_x86-64/0/DataFormats/common/include/CommonDataFormat/AbstractRef.h:69:32: error: argument may not have 'void' type
/sw/SOURCES/O2/13310-slc8_x86-64/0/DataFormats/common/include/CommonDataFormat/AbstractRef.h:73:31: error: argument may not have 'void' type
/sw/SOURCES/O2/13310-slc8_x86-64/0/DataFormats/common/include/CommonDataFormat/AbstractRef.h:77:26: error: argument may not have 'void' type
/sw/SOURCES/O2/13310-slc8_x86-64/0/DataFormats/common/include/CommonDataFormat/AbstractRef.h:80:30: error: argument may not have 'void' type
/sw/SOURCES/O2/13310-slc8_x86-64/0/DataFormats/common/include/CommonDataFormat/AbstractRef.h:90:10: error: field has incomplete type 'o2::dataformats::AbstractRef<25, 5, 2>::Base_t' (aka 'void')
/sw/SOURCES/O2/13310-slc8_x86-64/0/DataFormats/Reconstruction/include/ReconstructionDataFormats/GlobalTrackID.h:89:79: error: use of undeclared identifier 'getSource'; did you mean 'Source'?
/sw/SOURCES/O2/13310-slc8_x86-64/0/DataFormats/Reconstruction/include/ReconstructionDataFormats/GlobalTrackID.h:90:77: error: too few arguments to function call, single argument 'i' was not specified
/sw/SOURCES/O2/13310-slc8_x86-64/0/DataFormats/Reconstruction/include/ReconstructionDataFormats/GlobalTrackID.h:103:63: error: use of undeclared identifier 'getSource'; did you mean 'Source'?
/sw/SOURCES/O2/13310-slc8_x86-64/0/DataFormats/Reconstruction/include/ReconstructionDataFormats/GlobalTrackID.h:105:45: error: use of undeclared identifier 'getIndex'
ninja: build stopped: subcommand failed.
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/13310/5f3709fa8fc0431180c66d76290deb9a60f1b21b/build_O2_fullCI/pretty.html).

--- a/DataFormats/common/include/CommonDataFormat/AbstractRef.h
+++ b/DataFormats/common/include/CommonDataFormat/AbstractRef.h
@@ -20,9 +20,7 @@
 #include "GPUCommonRtypes.h"
 #include "GPUCommonTypeTraits.h"
 
-namespace o2
-{
-namespace dataformats
+namespace o2::dataformats
 {
 
 template <int NBIdx, int NBSrc, int NBFlg>
@@ -32,8 +30,19 @@ class AbstractRef
   static constexpr auto MVAR()
   {
     static_assert(NBIT <= 64, "> 64 bits not supported");
-    typename std::conditional<(NBIT > 32), uint64_t, typename std::conditional<(NBIT > 16), uint32_t, typename std::conditional<(NBIT > 8), uint16_t, uint8_t>::type>::type>::type tp = 0;
-    return tp;
+    if constexpr (NBIT > 32) {
+      uint64_t tp = 0;
+      return tp;
+    } else if constexpr (NBIT > 16) {
+      uint32_t tp = 0;
+      return tp;
+    } else if constexpr (NBIT > 8) {
+      uint16_t tp = 0;
+      return tp;
+    } else {
+      uint8_t tp = 0;
+      return tp;
+    }
   }
 
  public:
@@ -87,7 +96,6 @@ class AbstractRef
   ClassDefNV(AbstractRef, 1);
 };
 
-} // namespace dataformats
-} // namespace o2
+} // namespace o2::dataformats
 
 #endif

